test/index.js

```diff
@@ -295,15 +296,10 @@ suite('functions', function() {
   });
 
   test('empty', function() {
-    eq(Z.empty(''), '');
-    eq(Z.empty('abc'), '');
-    eq(Z.empty([]), []);
-    eq(Z.empty([1, 2, 3]), []);
-    eq(Z.empty({}), {});
-    eq(Z.empty({x: 1, y: 2, z: 3}), {});
-    eq(Z.empty(Identity('abc')), Identity(''));
+    eq(Z.empty(String), '');
+    eq(Z.empty(Array), []);
+    eq(Z.empty(Object), {});
```

Comment:
I've updated `Z.empty` to take a type representative, so `Identity('abc')` is no longer a valid value. This is in line with fantasyland/fantasy-land#180, which will preclude the Identity type from satisfying the requirements of Monoid.

Initially I thought we should support both `Identity` and `Identity('abc')` by requiring an argument of type `Monoid m => { fantasy-land/empty :: () -> m }`, but we'd also need to support `Array` and `[1, 2, 3]`. The description of `Z.of` would be quite complex, and the type signature would not tell the full story.

`Monoid m => TypeRep m` is accurate and compatible with fantasyland/fantasy-land#176.